src/views/steps/subviews/service.coffee

```diff
@@ -0,0 +1,83 @@
+{ItemView} = require 'backbone.marionette'
+_ = require 'underscore'
+
+ModalView = require '../../modal'
+
+module.exports = class ServiceView extends ItemView
+    template: require '../../templates/service'
+
+    className: 'service'
+
+    ui:
+        connect: '[role=connect]'
+
+    events:
+      'click [role=connect]': 'onConnect'
+
+
+    initialize: (options) ->
+        @service = options.service
+        @intent = options.intent
+
+        @onIntentStart = () =>
+            if typeof options.onIntentSuccess is 'function'
+                options.onIntentStart()
+
+        @onIntentSuccess = (doc) =>
+            if typeof options.onIntentSuccess is 'function'
+                options.onIntentSuccess doc
+
+        @onIntentError = (error) =>
+            if typeof options.onIntentError is 'function'
+                options.onIntentError error
+
+        @onIntentEnd = () =>
+            if typeof options.onIntentEnd is 'function'
+                options.onIntentEnd()
+
+
+    serializeData: () ->
+        _.extend super,
+            @service
+
+
+    onConnect: () ->
+        @onIntentStart()
+        @setBusy true
+
+        modal = new ModalView \
+            onHide: () =>
+                @onIntentEnd()
+                @setBusy false
+
+
+        cozy.client.intents
+            .create @intent.action, @intent.type
+            .start modal.getContentWrapper()
+            .then (doc) =>
+                modal.hide()
+                @onIntentSuccess doc
+            .catch (error) =>
+                modal.hide()
+                @onIntentError error
+            .then () =>
+                @setBusy false
+
+
+    setBusy: (busy) ->
+        if busy
+            @disableConnect()
+            @ui.connect.attr 'aria-busy', true
```

Comment:
In addition, I do not understand why you set `aria-busy` on the connect button, regarding its specification : https://www.w3.org/TR/wai-aria/states_and_properties#aria-busy. Busy should be set on the modal before `getContentWrapper` until `@onIntentSuccess` or `@onIntentError`, shouldn’t it ? Not shure of that last sentence.